src/main/java/com/github/dariobalinzo/elastic/ElasticIndexMonitorThread.java

```diff
@@ -93,7 +94,7 @@ private synchronized boolean updateIndexes() {
       return false;
     }
 
-    if (!indexes.equals(this.indexes)) {
+    if (!listEqualsIgnoreOrder(indexes, this.indexes)) {
```

Comment:
maybe it is easier here to just use Set instead of converting lists to sets for comparison.
e.g. might be directly a `Set<String> `and `elasticRepository.catIndices(this.prefix); `is converted to a set before the assignement.

In this way we can avoid the extra method that maybe breaks the single responsability of the class (`listEqualsIgnoreOrder` in fact may be used in other places etc.. so maybe in this case it is just better to use equals on sets)